lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
@serbanmaerean Do these other flags need to be listed? I don't think users will want to change `openmp_flag`, `pic_flag`, or `cxx11_flag`, will they?